### ISSUE TITLE
Phase 2: Train the Controls-v2 Player with gradient-based RL

### ISSUE THREAD

**Issue by rhythmatician** [rhythmatician/FractalSync#132]
Part of #130 and #77.

Coordinates with #104 (PlayerPolicy), #107 (Controls v2), #108/#109 (observation/audio contracts), #110 (reset distribution), #120 (authoritative manifold Physics), #95 (Julia presentation state), #111 (spectator/evaluation), #82 (native Shore crossing behavior), #128 (later entrainment reward if evidence-gated), and #131 (optional pretrained initialization).

- **Phase:** 2 — policy learning
- **Purpose:** learn how to act through Controls v2 from behavioral consequences in the destination dynamical system
- **Default optimization:** gradient-based RL, initially PPO or another justified actor-critic/policy-gradient baseline
- **Environment:** Rust-authoritative black-box transition system by default
- **Dependency nuance:** random-initialized RL may proceed without #131; only the pretrained-init arm depends on #131

## Goal

Build and validate the closed-loop destination policy-training path:

```text
audio tick t
  -> AudioActivation_t
  -> PlayerObservation_t
  -> causal PlayerPolicy(h_{t-1})
  -> ControlsV2_t
  -> Rust-authoritative manifold Physics
  -> Rust-authoritative Julia presentation integration
  -> world/presentation state_{t+1}
  -> behavioral reward / return / advantage
  -> gradient-based RL update
  -> next canonical tick
```

This phase answers:

> **Given what the Player understands and its current state, what should it do?**

There is generally no supervised target saying what throttle, steer, grip, impulse, Julia zoom, rotation, or ColorIntent value should be emitted at a specific tick.

## Core invariant

The Player can affect the world only through the canonical Controls-v2 action surface.

Forbidden trainer shortcuts include:

- direct target `c`;
- direct `x/y` acceleration;
- direct Mandelbrot `sigma` / `v_sigma` output;
- music-derived Physics force/potential/permeability changes;
- direct state mutation outside Rust-owned Controls interpretation;
- legacy `(s, alpha, omega_scale, band_gates)` as the destination action head.

## Gradient-based RL does not imply differentiable Physics

The baseline environment is an ordinary black-box Rust transition system:

```text
policy parameters theta
    -> sampled Controls
    -> Rust trajectory
    -> rewards / returns / advantages
    -> policy-gradient or actor-critic update of theta
```

PPO/policy-gradient-style learning requires gradients through the policy/log-probability/value model, not through the environment transition itself.

Therefore:

- Rust Physics remains the forward authority;
- no Python differentiable Physics mirror is required for baseline training;
- no renderer gradient is required;
- do not weaken/change forward mechanics to make them easier to differentiate.

If a later algorithm demonstrates a concrete need for environment gradients, treat that as a separate experiment under ADR 0001/#119, preferring Rust forward + explicit custom/surrogate backward.

## Inputs / authority

Consume rather than redefine:

- #109 `AudioActivation/1`;
- #108 `PlayerObservation`;
- #104 selected causal PlayerPolicy and hidden-state semantics;
- #107 Controls-v2 schema and interpretation;
- #120 manifold Physics;
- #110 reset-state distribution;
- #95/#107 Julia presentation integration;
- canonical sample-clock analysis/Physics timing;
- ADR 0001/#119 Rust-first semantics.

Do not create trainer-local copies of these contracts.

## Initialization arms

Phase 2 must support at least two matched initialization conditions:

### A. Random initialization baseline

```text
random trunk/policy initialization
    -> RL
```

This arm has no dependency on #131 and establishes whether the RL problem is learnable without representation pretraining.

### B. Phase-1 pretrained initialization

```text
#131 pretrained causal trunk
    -> attach/enable Controls-v2 policy head
    -> RL fine-tuning
```

The checkpoint must load without semantic/schema adaptation.

Optional later ablations may compare:

```text
pretrained trunk frozen initially
pretrained trunk fine-tuned immediately
pretrained -> RL + retained auxiliary losses
```

Retained auxiliary losses during RL are an explicit ablation, not part of the baseline definition.

## RL baseline

Start with a conventional, auditable algorithm rather than a bespoke optimizer.

Initial default:

```text
PPO
```

or another actor-critic/policy-gradient method if implementation evidence makes it a better fit.

Freeze/document:

- stochastic action distribution for each Controls-v2 field;
- bounded/continuous/discrete action semantics;
- value-function architecture;
- discount / GAE semantics;
- rollout horizon;
- recurrent-state handling;
- minibatch/epoch/update semantics;
- entropy/exploration treatment;
- gradient clipping / optimizer configuration;
- checkpoint/resume semantics.

Do not silently let algorithm convenience redefine Controls semantics.

## Chronology

At canonical tick `t`:

```text
O_t = observation(audio_{<=t}, world_t, julia_view_t)

(h_t, controls_t) = PlayerPolicy(O_t, h_{t-1})

(world_{t+1}, julia_view_{t+1})
    = step_destination_state(world_t, julia_view_t, controls_t, dt)
```

Requirements:

- one authoritative world-state transition per canonical analysis/Physics tick;
- no render-frame `dt`;
- future audio cannot affect earlier observations/actions;
- chunk/minibatch boundaries cannot silently reset forward recurrent state;
- evaluation and training use the same forward semantics.

## Reset distribution

Consume #110 directly.

Each rollout must preserve enough metadata to reproduce:

- `c0`, Map/Physics versions;
- velocity/Momentum, `K0`, `E0`;
- scale/reset band;
- Julia presentation state;
- Player hidden state initialization;
- sampler version / seed;
- song/excerpt/stream epoch.

Training may use scale/energy-stratified starts. Evaluation must separately report ordinary-start navigation/generalization.

## Reward ledger

Behavioral reward is distinct from Phase-1 teacher losses.

Keep each term separately logged/versioned, including where retained:

```text
basic driving / competence
continuity / chatter
control / impulse cost
Shore / topology timing objectives
roughness (#105 if retained)
entrainment (#128 only after evidence gate)
other explicit visual/presentation objectives
```

Do not collapse reward diagnostics into one scalar for debugging.

Any auxiliary teacher losses retained from Phase 1 during RL must be separately logged and ablated.

## Exploration and action distributions

Controls v2 contains bounded semantically different action channels. The policy distribution should respect those meanings.

At minimum verify:

- continuous throttle/steering/grip/view deltas stay within Rust-owned ranges;
- brake/dissipation cannot become an accidental energy source through action decoding;
- impulses/taps preserve their bounded/sparse semantics rather than becoming constant Gaussian chatter;
- circular presentation quantities use the canonical delta/state conventions;
- sampled raw action, log probability, and effective Rust-applied action can be inspected in #111.

## Evaluation

Primary comparison:

```text
RL from random initialization
vs
RL from #131 pretrained trunk
```

Use matched compute/seeds/reward/reset conditions where practical.

Measure:

- sample efficiency;
- reward learning curves by decomposed term;
- stability across seeds;
- held-out song/performance behavior;
- transition preparation and timing;
- scale/realm/energy coverage;
- native Shore-crossing competence;
- chatter / pathological limit cycles;
- action/impulse economy;
- eyes-on musical coherence in #111.

Do not call Phase 1 successful merely because it predicts teachers better; transfer to this phase is the decisive test.

## Deterministic spectator path

For evaluation, freeze:

```text
model checkpoint
song/excerpt
reset seed/state
contract versions
reward/evaluator version
```

and generate a reproducible chronological rollout consumable by #111.

The exact actions/snapshots used for metrics must be replayable for eyes-on inspection.

## Required tests

At minimum:

1. **Causality:** future audio cannot change earlier observations/actions/state.
2. **Chronology:** exactly one Rust destination state transition occurs per canonical tick.
3. **Controls-only action:** the policy cannot mutate world/presentation state except through Controls v2.
4. **Rust authority:** baseline RL requires no undeclared differentiable Physics forward mirror.
5. **Reset reproducibility:** same model/audio/reset metadata reproduces evaluation trajectory within tolerance.
6. **Chunk/recurrent invariance:** inference/evaluation chunking preserves forward hidden/world state.
7. **Action bounds/log-prob consistency:** sampled actions, transformed bounded actions, and PPO log probabilities correspond correctly.
8. **Impulse semantics:** taps/impulses cannot become repeated frame-level noise through distribution decoding.
9. **Checkpoint compatibility:** #131 pretrained trunk loads without observation/state schema adaptation.
10. **Random-init independence:** the baseline RL arm runs without requiring a Phase-1 checkpoint.
11. **Presentation parity:** Julia view-state evolution matches runtime semantics.
12. **Spectator reproducibility:** frozen evaluation rollouts replay exactly enough for #111 diagnosis.

## Acceptance

- [ ] A closed-loop destination RL trainer exists over exact AudioActivation/PlayerObservation/Controls-v2/Rust-Physics semantics.
- [ ] A documented gradient-based RL baseline is selected and runs without differentiable Physics.
- [ ] Random-initialized training is supported independently of #131.
- [ ] #131 pretrained checkpoints can initialize the same policy/trunk contract without adapters.
- [ ] Controls are the only learned action boundary.
- [ ] Hidden-state, rollout, reset, and canonical-time semantics are explicit and reproducible.
- [ ] Reward terms are decomposed/versioned and inspectable.
- [ ] At least one small Player demonstrably learns nontrivial Controls-v2 behavior.
- [ ] Random-init vs pretrained-init is compared under matched conditions.
- [ ] Held-out deterministic evaluation can be replayed in #111 with exact action/state provenance.
- [ ] No legacy orbit-control/product-head semantics are required by the destination trainer.
- [ ] No Python differentiable forward mirror is introduced merely because the optimizer is gradient-based.

## Non-goals

- Do not perform representation pretraining here; #131 owns it.
- Do not choose final AudioActivation/PlayerObservation/PlayerPolicy architecture here.
- Do not redesign Controls or Physics.
- Do not make Physics music-aware.
- Do not require differentiable environment dynamics.
- Do not implement neuroevolution/QD as the baseline in this ticket.
- Do not implement #128 entrainment reward before its evidence gate.
- Do not preserve legacy orbit-control training semantics for convenience.

**Comment by rhythmatician** [comment]
Add **#150** as a pre-full-RL research gate/ablation source. The new question is not whether PPO itself has a convergence proof on FractalSync, but whether the environment we hand to PPO is well conditioned.

Before claiming that scale-relative geometry improves learnability, compare matched training/optimization conditions under:

```text
Euclidean I
vs
scale-only rho^-2 I
vs
full G.
```

Also compare finite-horizon Controls response for at least basic navigation and rider-only vs rider+probe rhythmic expression. #150 defines the local conditioning diagnostics; this ticket should consume the result when selecting experiments and report sample efficiency/stability accordingly.

Do not add a differentiable Physics requirement. The conditioning analysis can be estimated offline/diagnostically around Rust-authoritative trajectories and then tested empirically with the existing black-box RL path.
